cmd/kubeadm/app/phases/certs/renewal/apirenewer_test.go

```diff
@@ -41,18 +43,27 @@ func TestAPIRenewer(t *testing.T) {
 		t.Fatalf("couldn't create CA: %v", err)
 	}
 
-	client := &fakecerts.FakeCertificatesV1beta1{
-		Fake: &k8stesting.Fake{},
-	}
+	client := fake.NewSimpleClientset()
 	certReq := getCertReq(t, caCert, caKey)
 	certReqNoCert := certReq.DeepCopy()
 	certReqNoCert.Status.Certificate = nil
-	client.AddReactor("get", "certificatesigningrequests", defaultReactionFunc(certReq))
+	client.PrependReactor("get", "certificatesigningrequests", defaultReactionFunc(certReq))
 	watcher := watch.NewFakeWithChanSize(3, false)
 	watcher.Add(certReqNoCert)
 	watcher.Modify(certReqNoCert)
 	watcher.Modify(certReq)
-	client.AddWatchReactor("certificatesigningrequests", k8stesting.DefaultWatchReactor(watcher, nil))
+	client.PrependWatchReactor("certificatesigningrequests", k8stesting.DefaultWatchReactor(watcher, nil))
+
+	// apirenewer depends on submitting CSRs without signerName, which requires creating via v1beta1.
+	// letting the create use v1beta1 but allowing the list/watch to use v1 requires simulating server-side conversion.
+	// instead, simulate v1 being unavailable so the list/watch falls back to v1beta1.
+	// once the apirenewer create switches to v1, this can be removed.
+	client.PrependReactor("*", "certificatesigningrequests", func(act k8stesting.Action) (bool, runtime.Object, error) {
+		if act.GetResource().Version == "v1" {
+			return true, nil, apierrors.NewNotFound(schema.GroupResource{Group: "certificates.k8s.io", Resource: "certificatesigningrequests"}, "")
+		}
+		return false, nil, nil
+	})
```

Comment:
i think we should just merge https://github.com/kubernetes/kubernetes/pull/90143 before this PR which removes cmd/kubeadm/app/phases/certs/renewal/apirenewer.go

Reply:
sgtm, I'll drop this file if that is in by the time the CSR chain of PRs gets here